ICSharpCode.Decompiler/DotNetCore/DotNetCorePathFinder.cs

```diff
@@ -112,5 +114,24 @@ static string FallbackToDotNetSharedDirectory(AssemblyNameReference name, string
 			}
 			return null;
 		}
+
+		static string FindDotNetExeDirectory()
+		{
+			char separator;
+			string dotnetExeName;
+			if (Environment.OSVersion.Platform == PlatformID.Unix) {
+				separator = ':';
+				dotnetExeName = "dotnet";
+			} else {
+				separator = ';';
+				dotnetExeName = "dotnet.exe";
+			}
+			foreach (var item in Environment.GetEnvironmentVariable("PATH").Split(separator)) {
+				var path = Environment.ExpandEnvironmentVariables(item);
```

Comment:
Why are you calling `ExpandEnvironmentVariables`? I don't think PATH is allowed to contain references to further envvars.
On the other hand, there's no guarantee that all elements in PATH are valid paths -- programs are supposed to ignore invalid elements. Beware of `Path.Combine()` throwing ArgumentException when PATH contains malformed elements!